config/initializers/rails_admin.rb

```diff
@@ -152,8 +152,16 @@
       end
     end
     edit do
-      field :uri
-      field :name
+      field :uri do
+        read_only do
+          true unless bindings[:view]._current_user.role == 'admin'
```

Comment:
there are a few ways descirbed to do this here: http://stackoverflow.com/questions/5921591/cancan-limiting-a-users-ability-to-set-certain-model-attributes-based-on-their
None of them are really very nice. Using cancan2 which never came out of beta, modifying ActiveRecord::Base, or just hiding the field like what was happening here. Since not having these fields editable was not a requirement on the ticket, I'll just leave them. Should this be an issue, we can reconsider the best approach.
In which case I would also suggest the option of just adding the DataProviderImage class to the cms and giving editors rights on that instead, if that's all they should be updating.